src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/ActionSvcClient.java

```diff
@@ -58,18 +234,84 @@ ActionPlanDTO getActionPlanBySelectors(String collectionExerciseId, boolean acti
+      final int priority,
+      final UUID actionPlanId)
+      throws RestClientException {
+    log.debug("Posting to action service to create action rule");
+    final UriComponents uriComponents =
+        restUtility.createUriComponents(appConfig.getActionSvc().getActionRulesPath(), null);
+
+    final ActionRulePostRequestDTO actionRulePostRequestDTO = new ActionRulePostRequestDTO();
+    actionRulePostRequestDTO.setName(name);
+    actionRulePostRequestDTO.setActionTypeName(actionTypeName);
+    actionRulePostRequestDTO.setDescription(description);
+    actionRulePostRequestDTO.setActionPlanId(actionPlanId);
+    actionRulePostRequestDTO.setTriggerDateTime(triggerDateTime);
+    actionRulePostRequestDTO.setPriority(priority);
+
+    final HttpEntity<ActionRulePostRequestDTO> httpEntity =
+        restUtility.createHttpEntity(actionRulePostRequestDTO);
+    final ActionRuleDTO createdActionRule =
+        restTemplate.postForObject(uriComponents.toUri(), httpEntity, ActionRuleDTO.class);
+    log.with("action_plan_id", actionPlanId)
+        .with("action_rule_id", createdActionRule.getId().toString())
+        .debug("Successfully posted to action service to create action rule");
+    return createdActionRule;
+  }
+
+  public ActionRuleDTO updateActionRule(
+      final UUID actionRuleId,
+      final String name,
+      final String description,
+      final OffsetDateTime triggerDateTime,
+      final int priority)
+      throws RestClientException {
+    final UriComponents uriComponents =
+        restUtility.createUriComponents(
+            appConfig.getActionSvc().getActionRulePath(), null, actionRuleId);
+
+    final ActionRulePutRequestDTO actionRulePutRequestDTO = new ActionRulePutRequestDTO();
+    actionRulePutRequestDTO.setName(name);
+    actionRulePutRequestDTO.setDescription(description);
+    actionRulePutRequestDTO.setTriggerDateTime(triggerDateTime);
+    actionRulePutRequestDTO.setPriority(priority);
+
+    final HttpEntity<ActionRulePutRequestDTO> httpEntity =
+        restUtility.createHttpEntity(actionRulePutRequestDTO);
+    return restTemplate
+        .exchange(uriComponents.toUri(), HttpMethod.PUT, httpEntity, ActionRuleDTO.class)
+        .getBody();
+  }
+
+  public List<ActionRuleDTO> getActionRulesForActionPlan(final UUID actionPlanId)
+      throws CTPException {
+    final UriComponents uriComponents =
+        restUtility.createUriComponents(
+            appConfig.getActionSvc().getActionRulesForActionPlanPath(), null, actionPlanId);
+
+    ResponseEntity<List<ActionRuleDTO>> response;
+    HttpEntity httpEntity = restUtility.createHttpEntity(null);
+
+    try {
+      response =
+          restTemplate.exchange(
+              uriComponents.toUri(),
+              HttpMethod.GET,
+              httpEntity,
+              new ParameterizedTypeReference<List<ActionRuleDTO>>() {});
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
```

Comment:
Are we expecting not to find something and successfully managing to continue processing? Feels a bit weird handling this explicitly otherwise.